src/globalComponents/typesAndClasses/SaleObject.js

```diff
@@ -6,13 +6,15 @@ class SaleObject {
 	amount: number;
 	sellerId: string;
 	saleId: string;
+	tankId: string;
```

Comment:
If sale objects are also used for item sales, then this should definitely be nullable. Otherwise you have a very confusing sentinel value...